addons/html_editor/static/src/main/separator_plugin.js

```diff
@@ -0,0 +1,86 @@
+import { _t } from "@web/core/l10n/translation";
+import { Plugin } from "../plugin";
+import { closestBlock } from "../utils/blocks";
+import { closestElement } from "../utils/dom_traversal";
+import { paragraphRelatedElements } from "../utils/dom_info";
+import { removeClass } from "@html_editor/utils/dom";
+
+export class SeparatorPlugin extends Plugin {
```

Comment:
Good idea to move the separator stuff in its own plugin! 👍 Next time you do so though, please make it in two commits. The first one that moves code without changes in it and the second one which makes the changes in the new file. Otherwise the diff of the commit does not show the difference and I have to manually compare the old code at the old place with the new code at the new place. In this case it wasn't a big deal because it was short, but just so you know. Thanks!